l1infotree/tree.go

```diff
@@ -84,16 +84,13 @@ func (mt *L1InfoTree) ComputeMerkleProof(gerIndex uint32, leaves [][32]byte) ([]
 		if len(leaves)%2 == 1 {
 			leaves = append(leaves, mt.zeroHashes[h])
 		}
-		if index%2 == 1 { //If it is odd
-			siblings = append(siblings, leaves[index-1])
-		} else { // It is even
-			if len(leaves) > 1 {
-				if index >= uint32(len(leaves)) {
-					// siblings = append(siblings, mt.zeroHashes[h])
-					siblings = append(siblings, leaves[index-1])
-				} else {
-					siblings = append(siblings, leaves[index+1])
-				}
+		if index >= uint32(len(leaves)) {
+			siblings = append(siblings, mt.zeroHashes[h])
+		} else {
```

Comment:
i'm not sure about this. If the index is 7 and the tree only have 3 leaves, this will break the function. I think you should check that you are in the even branch first